tests/SpecificationTests.cs

```diff
@@ -0,0 +1,52 @@
+using Xunit;
+using heitech.efXt;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+
+namespace tests
+{
+    public class SpecificationTests
+    {
+        [Fact]
+        public async Task Specification_works_with_List()
+        {
+            // Given
+            var source = Enumerable.Range(0, 14).Select(x => new TextContainer()).AsQueryable();
+            var spec = new FindNameEqMimiSpecification();
+
+            // When
+            var listResult = await spec.RunAsync(source);
+            var singleResult = await spec.RunSingleAsync(source);
+
+            // Then
+            listResult.Should().HaveCount(7);
+            singleResult.Should().NotBeNull();
```

Comment:
Perhaps we additionally want to assert that the `Name` is `"Mimi"` indeed:
`singleResult.Name.Should().Be("Mimi");`